### ISSUE TITLE
fastparquet/Dask 0.15.4 

### ISSUE THREAD

**Issue by apiszcz** [dask/fastparquet#234]
I tested Dask 0.15.4 (git pull) perhaps it is not ready yet. However, fastparquet (last modified version that works with Dask 0.15.3) does not work with latest gitpull Dask 0.15.4. Stack track follows. Retest with Dask 0.15.3 works and no stacktrace.

```
DASK 0.15.4
  File "lib\_test\test.py", line 664, in <module>
    main()
  File "lib\_test\test.py", line 655, in main
    ovdm.filter(oc,ol,ou,src,modes)
  File "lib\_test\test.py", line 430, in filter
    df = ddf.read_parquet(pfp, columns=columns, filters=fd['parquetfilter']).compute()
  File "C:\g\test\lib\dask\dataframe\io\parquet.py", line 293, in read_parquet
    categories=categories, index=index)
  File "C:\g\test\lib\dask\dataframe\io\parquet.py", line 63, in _read_fastparquet
    rgs = [rg for rg in pf.row_groups if
  File "C:\g\test\lib\dask\dataframe\io\parquet.py", line 65, in <listcomp>
    not(fastparquet.api.filter_out_cats(rg, filters))]
  File "C:\g\test\lib\fastparquet\api.py", line 712, in filter_out_cats
    out = filter_val(op, val, v0, v0)
  File "C:\g\test\lib\fastparquet\api.py", line 736, in filter_val
    if op in ['==', '>='] and val > vmax:
TypeError: unorderable types: int() > str()
```

**Comment by mrocklin** [comment]
I think that @martindurant pushed out a new release of fastparquet to help deal with this.  This was definitely an issue on our side though.  We should be testing more diligently on both latest release as well as git-master in our CI systems.

**Comment by apiszcz** [comment]
​Right, we have had a bit of git-dialog, concerning fastparquet on windows.
So I tried the latest this morning with Dask 0.15.4 and am seeing issues
continue, they may be different, apparently the separator interpretation
​is part of the issue.

**Comment by apiszcz** [comment]
I just tested fastparquet 0.1.3 with DASK 0.15.2 successfully.
I'll try the latest DASK next.

**Comment by apiszcz** [comment]
​DASK (github download), fastparquet 0.1.3 (github download)
​
Traceback (most recent call last):
  File "lib\_testdm\testdm.py", line 668, in <module>
    main()
  File "lib\_testdm\testdm.py", line 659, in main
    ovdm.filter(oc,ol,ou,src,modes)
  File "lib\_testdm\testdm.py", line 434, in filter
    df = ddf.read_parquet(pfp, columns=columns,
filters=fd['parquetfilter']).compute()
  File
"D:\p\test\test_v0045a_20171008_010853\lib\dask\dataframe\io\parquet.py",
line 293, in read_parquet
    categories=categories, index=index)
  File
"D:\p\test\test_v0045a_20171008_010853\lib\dask\dataframe\io\parquet.py",
line 63, in _read_fastparquet
    rgs = [rg for rg in pf.row_groups if
  File
"D:\p\test\test_v0045a_20171008_010853\lib\dask\dataframe\io\parquet.py",
line 65, in <listcomp>
    not(fastparquet.api.filter_out_cats(rg, filters))]
  File "D:\p\test\test_v0045a_20171008_010853\lib\fastparquet\api.py", line
711, in filter_out_cats
    out = filter_val(op, val, v0, v0)
  File "D:\p\test\test_v0045a_20171008_010853\lib\fastparquet\api.py", line
735, in filter_val
    if op in ['==', '>='] and val > vmax:
TypeError: unorderable types: int() > str()

**Comment by mrocklin** [comment]
Can you provide a reproducible example?

**Comment by apiszcz** [comment]
```python
import sys
import dask
from dask import dataframe as dd
import fastparquet as fp
import pandas as pd
import numpy as np
print('dask',dask.__version__,'fastparquet',fp.__version__)
df=pd.DataFrame(np.random.randint(0,100,(5,3)),columns=list('abc'))
fp.write('__fpdasktest.parq',df,file_scheme='hive',partition_on=['a'],compression='SNAPPY')

df2=dd.read_parquet('__fpdasktest.parq', filters=[('a','>',10)]).compute()
print(df,'\n',df2)

from dask154 import dataframe as dd
df3=dd.read_parquet('__fpdasktest.parq', filters=[('a','>',10)]).compute()
print(df3)

```

    dask 0.15.2 fastparquet 0.1.2
        a   b   c
    0  62  42  42
    1  73  98  27
    2  43  30  91
    3  27  58   4
    4  56  72  20 
         b   c   a
    0  72  20  56
    0  30  91  43
    0  58   4  27
    0  42  42  62
    0  98  27  73
    


    ---------------------------------------------------------------------------

    TypeError                                 Traceback (most recent call last)

    <ipython-input-22-9c09f1d37407> in <module>()
         13 
         14 from dask154 import dataframe as dd
    ---> 15 df3=dd.read_parquet('__fpdasktest.parq', filters=[('a','>',10)]).compute()
         16 print(df3)
    

    C:\g\via\lib\dask154\dataframe\io\parquet.py in read_parquet(path, columns, filters, categories, index, storage_options, engine)
        291         return _read_fastparquet(fs, paths, file_opener, columns=columns,
        292                                  filters=filters,
    --> 293                                  categories=categories, index=index)
        294     else:
        295         return _read_pyarrow(fs, paths, file_opener, columns=columns,
    

    C:\g\via\lib\dask154\dataframe\io\parquet.py in _read_fastparquet(fs, paths, myopen, columns, filters, categories, index, storage_options)
         61     name = 'read-parquet-' + tokenize(pf, columns, categories)
         62 
    ---> 63     rgs = [rg for rg in pf.row_groups if
         64            not(fastparquet.api.filter_out_stats(rg, filters, pf.schema)) and
         65            not(fastparquet.api.filter_out_cats(rg, filters))]
    

    C:\g\via\lib\dask154\dataframe\io\parquet.py in <listcomp>(.0)
         63     rgs = [rg for rg in pf.row_groups if
         64            not(fastparquet.api.filter_out_stats(rg, filters, pf.schema)) and
    ---> 65            not(fastparquet.api.filter_out_cats(rg, filters))]
         66 
         67     if index is False:
    

    C:\g\via\lib\fastparquet\api.py in filter_out_cats(rg, filters, sep)
        710                 v0 = val_to_num(v)
        711 
    --> 712             out = filter_val(op, val, v0, v0)
        713             if out is True:
        714                 return True
    

    C:\g\via\lib\fastparquet\api.py in filter_val(op, val, vmin, vmax)
        736         if op in ['==', '>='] and val > vmax:
        737             return True
    --> 738         if op == '>' and val >= vmax:
        739             return True
        740         if op == 'in' and min(val) > vmax:
    

    TypeError: unorderable types: int() >= str()

**Comment by apiszcz** [comment]
I changed line 375 in dask\dataframe\io\parquet.py to  sep = '/'
same error when I try to use dask 0.15.4 (or git)




```
import sys
import dask
from dask import dataframe as dd
import fastparquetgit as fp
import pandas as pd
import numpy as np
print('dask',dask.__version__,'fastparquet',fp.__version__)
​
df=pd.DataFrame(np.random.randint(0,100,(5,3)),columns=list('abc'))
fp.write('__fpdasktest.parq',df,file_scheme='hive',partition_on=['a'],compression='SNAPPY')
​
df2=dd.read_parquet('__fpdasktest.parq', filters=[('a','>',10)]).compute()
print(df,'\n',df2)
​
from dask154 import dataframe as dd
df3=dd.read_parquet('__fpdasktest.parq', filters=[('a','>',10)]).compute()
print(df3)
​



dask 0.15.2 fastparquet 0.1.3
    a   b   c
0  23  46  92
1  98  89   7
2  14  72  63
3  76  98  24
4  95  50  40 
     b   c   a
0  89   7  98
0  50  40  95
0  98  24  76
0  72  63  14
0  46  92  23

---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-8-64445e66143b> in <module>()
     14 
     15 from dask154 import dataframe as dd
---> 16 df3=dd.read_parquet('__fpdasktest.parq', filters=[('a','>',10)]).compute()
     17 print(df3)

C:\g\via\lib\dask154\dataframe\io\parquet.py in read_parquet(path, columns, filters, categories, index, storage_options, engine)
    291         return _read_fastparquet(fs, paths, file_opener, columns=columns,
    292                                  filters=filters,
--> 293                                  categories=categories, index=index)
    294     else:
    295         return _read_pyarrow(fs, paths, file_opener, columns=columns,

C:\g\via\lib\dask154\dataframe\io\parquet.py in _read_fastparquet(fs, paths, myopen, columns, filters, categories, index, storage_options)
     61     name = 'read-parquet-' + tokenize(pf, columns, categories)
     62 
---> 63     rgs = [rg for rg in pf.row_groups if
     64            not(fastparquet.api.filter_out_stats(rg, filters, pf.schema)) and
     65            not(fastparquet.api.filter_out_cats(rg, filters))]

C:\g\via\lib\dask154\dataframe\io\parquet.py in <listcomp>(.0)
     63     rgs = [rg for rg in pf.row_groups if
     64            not(fastparquet.api.filter_out_stats(rg, filters, pf.schema)) and
---> 65            not(fastparquet.api.filter_out_cats(rg, filters))]
     66 
     67     if index is False:

C:\g\via\lib\fastparquet\api.py in filter_out_cats(rg, filters, sep)
    709             else:
    710                 v0 = val_to_num(v)
--> 711             out = filter_val(op, val, v0, v0)
    712             if out is True:
    713                 return True

C:\g\via\lib\fastparquet\api.py in filter_val(op, val, vmin, vmax)
    735         if op in ['==', '>='] and val > vmax:
    736             return True
--> 737         if op == '>' and val >= vmax:
    738             return True
    739         if op == 'in' and min(val) > vmax:

TypeError: unorderable types: int() >= str()

```

**Comment by martindurant** [comment]
Fixed in #271
